variables.tf

```diff
@@ -91,6 +91,21 @@ variable "deployment_circuit_breaker" {
   })
 }
 
+variable "deployment_failure_detection_alarms" {
+  default = {
+    enable      = false
+    rollback    = false
+    alarm_names = []
+  }
+  description = "Deployment circuit breaker configuration."
+  type = object({
+    enable      = optional(bool, true)
```

Comment:
As this is a new pattern let me explain what this is doing. 
It defines the enable parameter as optional. If the user does not specify the object at all this will not be applied (see default above).

If the user defines the object then the enable value within that object is set to true per default. Which is in my opionon the desired behaviour. It still can be set to false if the user wants to disable the feature again